src/app.js

```diff
@@ -51,48 +51,59 @@ const isIdNum = (req, res, next) => {
 }
 
 // checks object keys are valid 
-const checkObjKeys = (req, res, next) => { 
-    const possibleKeys = ["name", "sex", "coat", "description", "breedId"]; 
+const checkObjKeys = (req, res, next) => {  
+
+    const possibleKeys = ["name", "sex", "coat", "description", "breed"];
+
+    function isKeyPresent (key) { 
+        console.log(`is ${key} present?`)
+        return possibleKeys.includes(key)
+    }
+
     const objToCheck = req.query;                // { name: "", sex: "", coat: ""}
-    arrayOfKeys = Object.keys(objToCheck);      // Eg: [ "name", "sex", "coat"]
-    arrayOfKeys.forEach((key) => { 
-        console.log(`checking key: ${key}`)
-        const index = possibleKeys.indexOf(key);    // alternative is to use .findByIndex(callback)
-        if (index < 0) {
-            let message =  `"${key}" is not a valid property`;
-            const newError = new Error(message);
-            newError.status = 404;
-            return next(newError); 
-        } else {
-        console.log(`key '${key}' verified`)
-        }
-    });
-    console.log('object successfully checked');
-    req.object = objToCheck; 
-    next();
+    console.log(objToCheck);
+    const arrayOfKeys = Object.keys(objToCheck);      // Eg: [ "name", "sex", "coat"]
+    console.log(arrayOfKeys);
+
+    if (arrayOfKeys.length < 1) {
+        const newError = new Error('no parameters provided');
+        newError.status = 404;
+        return next(newError);
+    }
+
+    if (!arrayOfKeys.every(isKeyPresent)) {
+        console.log(`One or more property is invalid`);
+        const newError = new Error('One or more property is invalid');
+        newError.status = 404;
+        return next(newError);
+    } else {
+        console.log('object keys checked'); 
+        req.object = objToCheck; 
+        next();
+    }
 }
 
 // checks the object values are valid 
 const checkObjValues = (req, res, next) => { 
     const objToCheck = req.query;      // { name: "", sex: "", coat: ""}
 
-    if (isInvalidString(objToCheck.name)) {     
+    if (objToCheck.name && isInvalidString(objToCheck.name)) { 
```

Comment:
`isInvalidString` could check whether the passed parameter is undefined/null, as this would avoid the need for an extra check.

Reply:
Done